### PR TITLE
Create DocsGPT.yml

### PR DESCRIPTION
**Pull request by dartpain**
Hi, Adding DocsGPT to the project, we have many issues developers can help with

Thank you!

### CODE REVIEW

**Comment by rifat01mahmud** [comment]
Generate the content for a YAML file that you can save as "DocsGPT.yml" on your local machine or server. Here's an example of what the content of a YAML file might look like:

```yaml
# DocsGPT Configuration File

model_name: gpt-3.5-turbo
temperature: 0.8
max_tokens: 100
top_p: 0.9
frequency_penalty: 0.0
presence_penalty: 0.6

# Add any additional parameters or settings you need
```

Feel free to modify the above template according to your specific needs. Once you have created the file and saved it as "DocsGPT.yml," you can use it to configure your application or script that interacts with the GPT-3.5 model.

**Comment by rifat01mahmud** [comment]
The given code is a GitHub Actions workflow file that runs a review process when a pull request is opened or synchronized with the base branch. Let's go through the workflow step by step:

```yaml
on:
  pull_request_target:
    types: [opened, synchronize]
```
This workflow is triggered when a pull request is opened or when an existing pull request is synchronized with the base branch.

```yaml
name: Review
```
The name of the workflow is "Review".

```yaml
jobs:
  reviewProjectChanges:
    name: Project Changes
    runs-on: ubuntu-latest
    timeout-minutes: 15
    steps:
```
The workflow defines a job named "Project Changes" that runs on an Ubuntu environment. The job has a timeout of 15 minutes.

```yaml
      - name: Checkout base branch
        uses: actions/checkout@v3
        with:
          ref: gh-pages
```
This step checks out the base branch (`gh-pages`) of the repository.

```yaml
      - name: Setup Ruby
        uses: ruby/setup-ruby@bc1dd263b68cb5626dbb55d5c89777d79372c484
        with:
          ruby-version: '3.2'
          bundler-cache: true
          cache-version: 0
```
This step sets up Ruby by using the `ruby/setup-ruby` action. It specifies the Ruby version as 3.2 and enables bundler caching with cache version 0.

```yaml
      - name: Run script to review changed projects
        id: run_script
        run: |
          MESSAGE=$(bundle exec ruby scripts/review_changes.rb) 
          echo 'message<<EOF' >> "$GITHUB_OUTPUT"
          echo "$MESSAGE" >> "$GITHUB_OUTPUT"
          echo 'EOF' >> "$GITHUB_OUTPUT"
        env:
          BASE_SHA: ${{ github.event.pull_request.base.sha }}
          HEAD_SHA: ${{ github.event.pull_request.head.sha }}
          PULL_REQUEST_NUMBER: ${{ github.event.pull_request.number }}
          GIT_REMOTE_URL: ${{ github.event.pull_request.head.repo.clone_url }}
          GITHUB_TOKEN: ${{ secrets.SHIFTBOT_TOKEN }}
```
This step runs a script (`scripts/review_changes.rb`) to review the changed projects. It captures the output message in the `MESSAGE` variable and writes it to a file named `$GITHUB_OUTPUT`. The script likely performs some kind of analysis or review on the changes introduced in the pull request.

The `env` section defines environment variables used within the script. It includes the base branch SHA, head branch SHA, pull request number, Git remote URL, and a GitHub token (`SHIFTBOT_TOKEN`) stored as a secret.

```yaml
      - uses: marocchino/sticky-pull-request-comment@f61b6cf21ef2fcc468f4345cdfcc9bda741d2343
        with:
          message: ${{ steps.run_script.outputs.message }}
          GITHUB_TOKEN: ${{ secrets.SHIFTBOT_TOKEN }}
```
This step uses the `marocchino/sticky-pull-request-comment` action to post a comment on the pull request. The comment message is taken from the output of the previous step (`steps.run_script.outputs.message`). It uses the same GitHub token (`SHIFTBOT_TOKEN`) as the previous step to authenticate and post the comment.

Overall, this workflow sets up Ruby, runs a script to review changes, and posts the review result as a comment on the pull request. It assumes the existence of a specific Ruby script (`scripts/review_changes.rb`) for performing the review.

**Comment by rifat01mahmud** [comment]
The given code is a GitHub Actions workflow file that runs a review process when a pull request is opened or synchronized with the base branch. Let's go through the workflow step by step:

```yaml
on:
  pull_request_target:
    types: [opened, synchronize]
```
This workflow is triggered when a pull request is opened or when an existing pull request is synchronized with the base branch.

```yaml
name: Review
```
The name of the workflow is "Review".

```yaml
jobs:
  reviewProjectChanges:
    name: Project Changes
    runs-on: ubuntu-latest
    timeout-minutes: 15
    steps:
```
The workflow defines a job named "Project Changes" that runs on an Ubuntu environment. The job has a timeout of 15 minutes.

```yaml
      - name: Checkout base branch
        uses: actions/checkout@v3
        with:
          ref: gh-pages
```
This step checks out the base branch (`gh-pages`) of the repository.

```yaml
      - name: Setup Ruby
        uses: ruby/setup-ruby@bc1dd263b68cb5626dbb55d5c89777d79372c484
        with:
          ruby-version: '3.2'
          bundler-cache: true
          cache-version: 0
```
This step sets up Ruby by using the `ruby/setup-ruby` action. It specifies the Ruby version as 3.2 and enables bundler caching with cache version 0.

```yaml
      - name: Run script to review changed projects
        id: run_script
        run: |
          MESSAGE=$(bundle exec ruby scripts/review_changes.rb) 
          echo 'message<<EOF' >> "$GITHUB_OUTPUT"
          echo "$MESSAGE" >> "$GITHUB_OUTPUT"
          echo 'EOF' >> "$GITHUB_OUTPUT"
        env:
          BASE_SHA: ${{ github.event.pull_request.base.sha }}
          HEAD_SHA: ${{ github.event.pull_request.head.sha }}
          PULL_REQUEST_NUMBER: ${{ github.event.pull_request.number }}
          GIT_REMOTE_URL: ${{ github.event.pull_request.head.repo.clone_url }}
          GITHUB_TOKEN: ${{ secrets.SHIFTBOT_TOKEN }}
```
This step runs a script (`scripts/review_changes.rb`) to review the changed projects. It captures the output message in the `MESSAGE` variable and writes it to a file named `$GITHUB_OUTPUT`. The script likely performs some kind of analysis or review on the changes introduced in the pull request.

The `env` section defines environment variables used within the script. It includes the base branch SHA, head branch SHA, pull request number, Git remote URL, and a GitHub token (`SHIFTBOT_TOKEN`) stored as a secret.

```yaml
      - uses: marocchino/sticky-pull-request-comment@f61b6cf21ef2fcc468f4345cdfcc9bda741d2343
        with:
          message: ${{ steps.run_script.outputs.message }}
          GITHUB_TOKEN: ${{ secrets.SHIFTBOT_TOKEN }}
```
This step uses the `marocchino/sticky-pull-request-comment` action to post a comment on the pull request. The comment message is taken from the output of the previous step (`steps.run_script.outputs.message`). It uses the same GitHub token (`SHIFTBOT_TOKEN`) as the previous step to authenticate and post the comment.

Overall, this workflow sets up Ruby, runs a script to review changes, and posts the review result as a comment on the pull request. It assumes the existence of a specific Ruby script (`scripts/review_changes.rb`) for performing the review.

**Comment by rifat01mahmud** [comment]
The given code is a GitHub Actions workflow file that runs a review process when a pull request is opened or synchronized with the base branch. Let's go through the workflow step by step:

```yaml
on:
  pull_request_target:
    types: [opened, synchronize]
```
This workflow is triggered when a pull request is opened or when an existing pull request is synchronized with the base branch.

```yaml
name: Review
```
The name of the workflow is "Review".

```yaml
jobs:
  reviewProjectChanges:
    name: Project Changes
    runs-on: ubuntu-latest
    timeout-minutes: 15
    steps:
```
The workflow defines a job named "Project Changes" that runs on an Ubuntu environment. The job has a timeout of 15 minutes.

```yaml
      - name: Checkout base branch
        uses: actions/checkout@v3
        with:
          ref: gh-pages
```
This step checks out the base branch (`gh-pages`) of the repository.

```yaml
      - name: Setup Ruby
        uses: ruby/setup-ruby@bc1dd263b68cb5626dbb55d5c89777d79372c484
        with:
          ruby-version: '3.2'
          bundler-cache: true
          cache-version: 0
```
This step sets up Ruby by using the `ruby/setup-ruby` action. It specifies the Ruby version as 3.2 and enables bundler caching with cache version 0.

```yaml
      - name: Run script to review changed projects
        id: run_script
        run: |
          MESSAGE=$(bundle exec ruby scripts/review_changes.rb) 
          echo 'message<<EOF' >> "$GITHUB_OUTPUT"
          echo "$MESSAGE" >> "$GITHUB_OUTPUT"
          echo 'EOF' >> "$GITHUB_OUTPUT"
        env:
          BASE_SHA: ${{ github.event.pull_request.base.sha }}
          HEAD_SHA: ${{ github.event.pull_request.head.sha }}
          PULL_REQUEST_NUMBER: ${{ github.event.pull_request.number }}
          GIT_REMOTE_URL: ${{ github.event.pull_request.head.repo.clone_url }}
          GITHUB_TOKEN: ${{ secrets.SHIFTBOT_TOKEN }}
```
This step runs a script (`scripts/review_changes.rb`) to review the changed projects. It captures the output message in the `MESSAGE` variable and writes it to a file named `$GITHUB_OUTPUT`. The script likely performs some kind of analysis or review on the changes introduced in the pull request.

The `env` section defines environment variables used within the script. It includes the base branch SHA, head branch SHA, pull request number, Git remote URL, and a GitHub token (`SHIFTBOT_TOKEN`) stored as a secret.

```yaml
      - uses: marocchino/sticky-pull-request-comment@f61b6cf21ef2fcc468f4345cdfcc9bda741d2343
        with:
          message: ${{ steps.run_script.outputs.message }}
          GITHUB_TOKEN: ${{ secrets.SHIFTBOT_TOKEN }}
```
This step uses the `marocchino/sticky-pull-request-comment` action to post a comment on the pull request. The comment message is taken from the output of the previous step (`steps.run_script.outputs.message`). It uses the same GitHub token (`SHIFTBOT_TOKEN`) as the previous step to authenticate and post the comment.

Overall, this workflow sets up Ruby, runs a script to review changes, and posts the review result as a comment on the pull request. It assumes the existence of a specific Ruby script (`scripts/review_changes.rb`) for performing the review.

File: _data/projects/DocsGPT.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+desc:  GPT-powered chat for documentation, chat with your documents 
+site: https://github.com/arc53/DocsGPT
+tags:
+- python
+- javascript
+- docker
+- automation
+upforgrabs:
+  name: good first issue
+  link: https://github.com/arc53/DocsGPT/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22
```
Comment (shiftkey) [inline _data/projects/DocsGPT.yml]:
```suggestion
  link: https://github.com/arc53/DocsGPT/labels/good%20first%20issue
```

**Pull request merged**